### PR TITLE
Add a bash script that automates the setup in the README.

### PR DESCRIPTION
**Pull request by JNProtzman**

### CODE REVIEW

**Comment by JNProtzman** [comment]
Fixes #29

File: setup/setup.sh [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+  SERVER_IP="0.0.0.0/32"
+fi
+
+read -rp "Please enter the external server address (default: 127.0.0.1): " EXTERNAL_ADDRESS
+if [ ! "$EXTERNAL_ADDRESS" ]; then
+  SERVER_IP="127.0.0.1"
+fi
+
+# No matter where we're calling this from, we're going to use
+# The root archon directory as our install location.
```
Comment (dcrodman) [inline setup/setup.sh:51]: Maybe this should be user-configurable?

File: setup/setup.sh [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+fi
+
+read -rp "Please enter the server address (default: 0.0.0.0/32): " SERVER_IP
+if [ ! "$SERVER_IP" ]; then
+  SERVER_IP="0.0.0.0/32"
+fi
+
+read -rp "Please enter the external server address (default: 127.0.0.1): " EXTERNAL_ADDRESS
+if [ ! "$EXTERNAL_ADDRESS" ]; then
+  SERVER_IP="127.0.0.1"
```
Comment (dcrodman) [inline setup/setup.sh:47]:
```suggestion
  EXTERNAL_ADDRESS="127.0.0.1"
```

File: setup/setup.sh [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+# No matter where we're calling this from, we're going to use
+# The root archon directory as our install location.
+SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+
+# Go to the root archon directory.
+pushd "$SCRIPT_DIR"/.. > /dev/null 2>&1
+
+SETUP_DIR="$(pwd)/setup"
+
+echo "Please edit the config file: $SETUP_DIR/config.yaml"
```
Comment (dcrodman) [inline setup/setup.sh:59]: Not sure I follow why this comment exists. Given this script is (rightly) copying the setup directory and editing the configs for which user input was collected, maybe this should be replaced by a message at the end saying to confirm settings before starting the server or something like that

File: setup/setup.sh [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+
+echo "Generating certificates..."
+./generate_cert --ip "$SERVER_IP" > /dev/null 2>&1
+echo "Done."
+
+echo "Adding account..."
+./account --config . add
+echo "Done."
+
+echo "Starting server..."
```
Comment (dcrodman) [inline setup/setup.sh:129]: Don't think this should actually start the server, most of these magic scripts just set everything up and then tell the user what next steps are. Just echoing "Start the server with..." is probably sufficient

File: setup/setup.sh [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+sed -i '' "s#$SEARCH#$REPLACE#" config.yaml
+
+# Edit key location
+SEARCH='ssl_key_file: "key.pem"'
+REPLACE="ssl_key_file: \"$(pwd)/key.pem\""
+sed -i '' "s#$SEARCH#$REPLACE#" config.yaml
+
+createdb "$DB_NAME"
+psql archondb -c "CREATE USER $ARCHON_USER WITH ENCRYPTED PASSWORD '$ARCHON_PASSWORD';"
+psql archondb -c "GRANT ALL ON ALL TABLES IN SCHEMA public TO $ARCHON_USER;"
```
Comment (dcrodman) [inline setup/setup.sh:115]:
```suggestion
psql $DB_NAME -c "CREATE USER $ARCHON_USER WITH ENCRYPTED PASSWORD '$ARCHON_PASSWORD';"
psql $DB_NAME -c "GRANT ALL ON ALL TABLES IN SCHEMA public TO $ARCHON_USER;"
```

**Review by dcrodman — APPROVED** [review]

**Pull request merged**